backend/editor/api.py

```diff
@@ -35,4 +57,214 @@ async def pong(response: Response):
     Check server health
     """
     pong = datetime.now()
-    return {"ping": "pong @ %s" % pong}
\ No newline at end of file
+    return {"ping": "pong @ %s" % pong}
+
+@app.get("/nodesfull")
+async def findAllNodes(response: Response):
+    """
+    Get all nodes within taxonomy
+    """
+    query = """
+        MATCH (n) RETURN n
+    """
+    result = session.run(query)
+    allNodes = [record for record in result]
+    return allNodes
+
+@app.get("/entry/{entry}")
+async def findOneEntry(response: Response, entry: str):
+    """
+    Get entry corresponding to id within taxonomy
+    """
+    query = """
+        MATCH (n:ENTRY) WHERE n.id = $id 
+        RETURN n
+    """
+    result = session.run(query, {"id": entry})

```

Comment:
It would be better to separate database access from http processing.

So it would be better to have a entries.py where you put:
- the session initialization / shutdown (but events handlers should be here, they just call the right function)
- all the requests as simple functions (eg, here: `get_entry(id)`)

It's always better to separate that kind of things:
- you can factor out something (here for example I imagine only one function get_node(id, type), and in findOneEntry, we can call `get_node(id, "ENTRY")° (BTW "ENTRY" could be in an enumeration)
- for testability (you can test db function on their own)
- as you code will grow you will have a better [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) (and this is a very important design pattern in my experience)